sbncode/FluxReader/DK2NuInterface.cxx

```diff
@@ -0,0 +1,291 @@
+
+    double dot = fFluxWindowDir1.Dot(fFluxWindowDir2);
+    if ( TMath::Abs(dot) > 1.0e-8 )
+      std::cout << "Dot product between window direction vectors was "
+		<< dot << "; please check for orthoganality"<<std::endl;
+
+    //overwrite dkmeta
+    fDkMeta->location.clear();
+    fDkMeta->location.resize(2);
+    fDkMeta->location[0].x=0;
+    fDkMeta->location[0].y=0;
+    fDkMeta->location[0].z=0;
+    TLorentzVector detVec;
+    User2BeamPos(TLorentzVector(0,0,0,0),detVec);
+    fDkMeta->location[1].x=detVec.Vect().X();
+    fDkMeta->location[1].y=detVec.Vect().Y();
+    fDkMeta->location[1].z=detVec.Vect().Z();
+
+    std::cout<<"Locations "<<std::endl;
+    for (unsigned int i=0;i<fDkMeta->location.size();i++) {
+      std::cout<<i<<"\t"<<fDkMeta->location[i].x<<"\t"
+	       <<fDkMeta->location[i].y<<"\t"
+	       <<fDkMeta->location[i].z<<std::endl;
+    }
+  }
+  void DK2NuInterface::User2BeamPos(const TLorentzVector& usrxyz,
+                                   TLorentzVector& beamxyz) const
+  {
+    beamxyz = fBeamRotInv*(usrxyz-fBeamZero);
+  }
+  void DK2NuInterface::Beam2UserPos(const TLorentzVector& beamxyz,
+                                    TLorentzVector& usrxyz) const
+  {
+    usrxyz = fBeamRot*beamxyz + fBeamZero;
+  }
+  void DK2NuInterface::Beam2UserP4 (const TLorentzVector& beamp4,
+                                    TLorentzVector& usrp4 ) const
+  {
+    usrp4 = fBeamRot*beamp4;
+  }
+
+    TVector3 DK2NuInterface::AnglesToAxis(double theta, double phi)
+  {
+    //theta,phi in rad
+    double xyz[3];
+    xyz[0] = TMath::Cos(phi)*TMath::Sin(theta);
+    xyz[1] = TMath::Sin(phi)*TMath::Sin(theta);
+    xyz[2] = TMath::Cos(theta);
+    // condition vector to eliminate most floating point errors
+    for (int i=0; i<3; ++i) {
+      const double eps = 1.0e-15;
+      if (TMath::Abs(xyz[i])   < eps ) xyz[i] =  0;
+      if (TMath::Abs(xyz[i]-1) < eps ) xyz[i] =  1;
+      if (TMath::Abs(xyz[i]+1) < eps ) xyz[i] = -1;
+    }
+    return TVector3(xyz[0],xyz[1],xyz[2]);
+  }
+
+  bool DK2NuInterface::FillMCFlux(Long64_t ientry, simb::MCFlux& flux)
+
+  {
+    if (!fDk2NuTree->GetEntry(ientry))
+      return false;
+
+    TLorentzVector x4beam=fFluxWindowBase+fRnd.Uniform()*fFluxWindowDir1+fRnd.Uniform()*fFluxWindowDir2;
+    double enu,wgt;
+    bsim::calcEnuWgt(fDk2Nu, x4beam.Vect(),enu,wgt);
+
+    TLorentzVector x4usr;
+    Beam2UserPos(x4beam,x4usr);
+
+    bsim::NuRay rndnuray=fDk2Nu->nuray[0];
+    fDk2Nu->nuray.clear();
+
+    TVector3 xyzDk(fDk2Nu->decay.vx,fDk2Nu->decay.vy,fDk2Nu->decay.vz);  // origin of decay
+    TVector3 p3beam = enu * (x4beam.Vect()-xyzDk).Unit();
+
+    //weight due to window being tilted with respect to beam direction
+    double tiltwgt = p3beam.Unit().Dot( fWindowNormal );
+    wgt*=tiltwgt;
+    //weight for the window area and divide by pi (since wgt returned by calcEnuWgt function is flux/(pi*m^2)
+    wgt*=fWindowArea/3.14159;
```

Comment:
`TMath::Pi()` or `M_PI`. And I am pretty confused by the unit convention that requires this....